src/west/app/main.py

```diff
@@ -860,21 +841,61 @@ def add_argument(self, *args, **kwargs):
         super().add_argument(*args, **kwargs)
 
     def error(self, message):
-        if (self.west_app and
-                self.west_app.mle and
-                isinstance(self.west_app.mle,
-                           ManifestVersionError) and
-                self.west_app.cmd):
-            self.west_app.cmd.die(mve_msg(self.west_app.mle))
+        if self.west_app and self.west_app.mle:
+            # If we have a known WestApp instance and the manifest
+            # failed to load, then try to specialize the error message
+            # to handle west-specific situations.
+
+            app = self.west_app
+            mle = self.west_app.mle
+            if app.cmd:
+                cmd = app.cmd
+            else:
+                # No app.cmd probably means that the user is
+                # running an extension command that they expected
+                # to work, but we don't know about because the
+                # import failed and thus we have no manifest to
+                # load extensions from.
+                #
+                # Just use the help command as a stand-in instead.
+                # We have to manually patch up its config
+                # attribute in order to do this outside of
+                # do_run().
+                cmd = app.builtins['help']
+                cmd.config = west.configuration.Configuration(
+                    topdir=app.topdir)
```

Comment:
I don't understand why this `.config` "patch-up" is needed sorry.

Reply:
Because we aren't inside do_run(), and thus configuration isn't accessible in the usual way. This is part of the parameter contract for WestCommand.config:

```
        '''Property for the config which was passed to run().

        If `do_run` was given a *config* kwarg, it is returned.
        Otherwise, a fatal error occurs.
        '''
```

We're not running a command, so we're not in do_run(). So we have to patch this up manually.

Reply:
OK, it looks like I don't really know the meaning of the English verb "to patch up".

I thought "to patch up" meant modifying an existing `.config` but you're saying this code is making a _missing_ one up? That makes more sense to me now.

So I would have written "make up a missing .config attribute" but you're the native speaker and I'm not so never mind!